frontend/src/features/public-form/utils/inputTransformation.ts

```diff
@@ -0,0 +1,237 @@
+import { format } from 'date-fns'
+
+import { BasicField, FormFieldDto } from '~shared/types/field'
+import {
+  AttachmentResponse,
+  CheckboxResponse,
+  FieldResponse,
+  HeaderResponse,
+  RadioResponse,
+  TableResponse,
+  TableRow,
+} from '~shared/types/response'
+import { removeAt } from '~shared/utils/immutable-array-fns'
+
+import { CHECKBOX_OTHERS_INPUT_VALUE } from '~templates/Field/Checkbox/CheckboxField'
+import { RADIO_OTHERS_INPUT_VALUE } from '~templates/Field/Radio/RadioField'
+import {
+  AttachmentFieldSchema,
+  BaseFieldOutput,
+  CheckboxFieldSchema,
+  DateFieldSchema,
+  EmailFieldSchema,
+  MobileFieldSchema,
+  RadioFieldSchema,
+  SectionFieldSchema,
+  SingleAnswerOutput,
+  TableFieldSchema,
+  VerifiableAnswerOutput,
+} from '~templates/Field/types'
+
+import {
+  validateAttachmentInput,
+  validateCheckboxInput,
+  validateDateInput,
+  validateRadioInput,
+  validateSingleAnswerInput,
+  validateTableInput,
+  validateVerifiableInput,
+} from './inputValidation'
+
+export class InputValidationError extends Error {
+  constructor(schemaId: string, input: unknown) {
+    super(`${schemaId}: 'Invalid input: ${JSON.stringify(input)}'`)
+    this.name = 'InputValidationError'
+  }
+}
+
+const pickBaseOutputFromSchema = <F extends FormFieldDto>(
+  schema: F,
+): BaseFieldOutput<F> => {
+  return {
+    _id: schema._id,
+    fieldType: schema.fieldType,
+    question: schema.title,
+  }
+}
+
+const transformToVerifiableOutput = <
+  F extends EmailFieldSchema | MobileFieldSchema,
+>(
+  schema: F,
+  input: unknown,
+): VerifiableAnswerOutput<F> => {
+  if (
+    !validateVerifiableInput(input) ||
+    (schema.isVerifiable && !input.signature)
+  ) {
+    throw new InputValidationError(schema._id, input)
+  }
+  return {
+    ...pickBaseOutputFromSchema(schema),
+    answer: input.value,
```

Comment:
nit: maybe we can consider this as "inheriting" from `SingleAnswer`?
```suggestion
    ...transformToSingleAnswerOutput(schema, input),
    answer: input.value,
```

Reply:
not possible, since `SingleAnswerInput === string`, which is not the case for `VerifiableInput`, which has the shape 
```ts
{ value: string, signature: string }
```